code/src/sixsq/nuvla/server/resources/spec/notification.cljc

```diff
@@ -0,0 +1,91 @@
+(ns sixsq.nuvla.server.resources.spec.notification
+  (:require
+    [clojure.spec.alpha :as s]
+    [sixsq.nuvla.server.resources.spec.common :as common]
+    [sixsq.nuvla.server.resources.spec.core :as core]
+    [sixsq.nuvla.server.util.spec :as su]
+    [spec-tools.core :as st]))
+
+
+(s/def ::message
+  (-> (st/spec ::core/nonblank-string)
+      (assoc :name "message"
+             :json-schema/type "string"
+             :json-schema/description "notification text"
+
+             :json-schema/order 30)))
+
+
+(s/def ::type
+  (-> (st/spec ::core/kebab-identifier)
+      (assoc :name "type"
+             :json-schema/description "notification type"
+
+             :json-schema/order 31)))
+
+
+(s/def ::content-unique-id
+  (-> (st/spec ::core/nonblank-string)
+      (assoc :name "content-unique-id"
```

Comment:
For compound attribute names, you probably want to define:
```
:json-schema/display-name "API key"
```

Reply:
Addressed in 6c6f792